### DISCUSSION TITLE
Programmatically login with custom logic

### DISCUSSION THREAD

**Discussion by netsinerji** [Budibase/budibase#7823] [General]
Coming from ASP.net world, I miss the opportunity of programmatically login any user with just one line of code.
I was using it with magic-links and other custom authentication scenarios such as self-made OTP.
Is it possible to do this in Budibase?
If not, can anybody kindly advise a work around? Such as using one of the supported auth methods and put my logic somewhere in that auth provider maybe?

**Comment by melohagan** [comment]
Hi @netsinerji 

I don't know of a way to programmatically determine auth within Budibase, however you can use [OpenID Connect](https://docs.budibase.com/docs/openid-connect) to build your own OAuth 2 workflow.
